File: Day02_OC加强_内存管理2/01-autoreleasepool的基本使用补充/13-分类的使用注意事项/Person.m
//
//  Person.m
//  01-autoreleasepool的基本使用补充
//
//  Created by SteveChiang on 15/12/22.
//  Copyright © 2015年 Apple.Inc. All rights reserved.
//

#import "Person.h"

@implementation Person

-(void)run
{
    NSLog(@"run...");
}

@end

@implementation Person (printNameAndAge)

-(void)printNameAndAge
{
    NSLog(@"name: %@ -- age:%d", _name, _age);
}

-(void)run
{
    NSLog(@"run run....");
}

@end

File: Day31_新知识_Swift入门/12_getter和setter方法的/12_getter和setter方法的/Person.swift

//
//  Person.swift
//  12_getter和setter方法的
//
//  Created by JiangDi on 16/5/27.
//  Copyright © 2016年 Google. All rights reserved.
//

import UIKit

class Person: NSObject {
    
    var _name: String?
    
    var name: String? {
        get {
            return _name
        }
        
        set {
            _name = newValue
        }
    }
    
    
    // 一般开发中的写法：
    var gender: String? {
        // 设置完gender的值的时候调用：
        didSet {
            print(gender)
        }
    }
    
    var age: Int {
        get {
            return 30
        }
    }
    
}
